src/pages/posts.js

```diff
@@ -0,0 +1,68 @@
+import React from "react"
+import { Link, graphql } from "gatsby"
+import styles from "./styles/posts.module.css"
+
+import Layout from "../components/layout"
+
+export default function Posts(props) {
+
+  const data = props.data.allMarkdownRemark;
+
+  return (
+  <Layout>
+
+    <div 
+    className={styles.header}
+    >
+        Hello Posts!
+    </div>
+
+    <div
+    className={styles.articleList}
+    >
+      <h3>
+        {data.totalCount} Posts
+      </h3>
+      {data.edges.map(({ node }) => (
```

Comment:
This is probably a place where I'd use argument destructuring to save on needing to type `node` over and over. Something like:

```js
export default ({ 
  data: { edges, excerpt }, 
  node: { fields, frontmatter, id }
}) => (
  <Layout>
    ...
    <Link
      ...
      key={id}
    >
    <h2>
      {frontmatter.title}
    </h2>
  </Layout>
)
```

Interested in @solomonhawk' thoughts here, though. My general rule is that once I'm destructuring around 7 values, it stops being a "neat trick" and starts getting confusing. For just a few values, though, it can be convenient to pluck them out of the object like this.